Sources/Implementation/Events/BatchEventBuilder.swift

```diff
@@ -25,13 +25,29 @@ class BatchEventBuilder {
     
     static func createImpressionEvent(config: ProjectConfig,
                                       experiment: Experiment,
-                                      varionation: Variation,
+                                      variation: Variation?,
                                       userId: String,
-                                      attributes: OptimizelyAttributes?) -> Data? {
+                                      attributes: OptimizelyAttributes?,
+                                      flagKey: String,
+                                      flagType: String) -> Data? {
+        
+        if (flagType == Constants.DecisionSource.rollout.rawValue || variation == nil) && !config.sendFlagDecisions() {
+            return nil
+        }
+
+        var variationId = ""
+        var variationKey = ""
+        if let tmpVariation = variation {
```

Comment:
why using it? 
variation?.Id should be fine.

Reply:
Value of optional type 'String?' must be unwrapped to a value of type 'String' before assigning it into the struct. This change is there since variation can now be nil too.